uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/calender-instance.js

```diff
@@ -0,0 +1,72 @@
+define([
+  'common/moment',
+  'leave-absences/shared/modules/models-instances',
+  'common/models/instances/instance',
+], function (moment, instances) {
+  'use strict';
+
+  instances.factory('CalenderInstance', [
+    'ModelInstance',
+    function (ModelInstance) {
+
+      /**
+       * This method checks whether a date matches the send type.
+       *
+       * @param {string} Date
+       * @param {string} Type of day
+       *
+       * @return {Boolean}
+       * Throws Error if date is not found in calenderData
+       */
+      function checkDate(date, dayType) {
+        var searchedDate = this.calenderData.find(function (data) {
+          return moment(data.date).isSame(moment(date));
+        });
+
+        if (!searchedDate) {
+          throw new Error("Date not found");
+        }
+
+        return searchedDate.type.name === dayType;
```

Comment:
If the instance is going to store the day types as list of option values (see my other comment about it), then maybe it would be better to test the `id` instead of the `name`? Not sure, just a thought